### DISCUSSION TITLE
Getting ControlNet processing parameters for a script

### DISCUSSION THREAD

**Discussion by JasonS09** [Mikubill/sd-webui-controlnet#1025] [Q&A] [answered]
Hello. I'm working on a script that requires accessing ControlNet parameters, which I think can't be found in StableDiffusionProcessing class. How can I do to access ControlNet parameters defined by user in the UI? Is it possible?

**Comment by ljleb** [comment, marked as answer]
Check the wiki: https://github.com/Mikubill/sd-webui-controlnet/wiki/API#update-existing-arguments

If you only need to read the values, you do not need to call `external_code.update_cn_script_in_processing`. You will receive a list of controlnet unit objects when calling `external_code.get_all_units_in_processing(p)`.

Note that your script must be invoked after controlnet, otherwise you will not get up-to-date values.

**Reply by JasonS09** [reply]
Thanks for the quick answer.
